akshya-intern

achaudhary591 opened the pull request:
created orderDetail entity , repository, service and controller

techeazy-consulting commented on src/main/java/com/techeazy/lmds/model/OrderDetail.java:
```diff
@@ -0,0 +1,59 @@
+    private long id;
+
+    private String recipientAddress;
+    private int itemQuantity;
```
Receipient address is not needed

achaudhary591 commented:
1.Added Lombok to reduce boiler plate and hibernate validator dependecy to make validation like making fields notnoll in entities
2. added client(entity, repo , service and controller)
   3.Mapped orderDetails to parcels (Manytoone)
4. Mapped Client to OrderDetail (onetoone)
5. Removed recipient's address from orderdetail

cpandey05 commented on src/main/java/com/techeazy/lmds/repo/ParcelRepository.java:
```diff
@@ -1,10 +1,20 @@
+    Optional<Parcel> findByTrackingId(String trackingId);
+
+    @Query("SELECT p FROM Parcel p WHERE p.orderDetail.client.id = :clientId AND p.status = :status")
+    List<Parcel> findByClientIdAndStatus(@Param("clientId") Long clientId, @Param("status") String status);
```
Is this query needed

achaudhary591 commented:
Added role based access control for two users -> ADMIN and USER, any endpoint contains "/admin/*" can only be accessed by admin only and those which have user , can be accessed by user, for test purpose you can check and hit endpoints of HelloResource controller.
